### DISCUSSION TITLE
return an empty value without an error

### DISCUSSION THREAD

**Discussion by electriquo** [TomWright/dasel#328] [Q&A] [answered]
if i am not mistaken, in the past dasel did not fail when accessing properties that do not exist and was returning an empty value, which is not the case for dasel version `2.3.4`. for instance: 

```shell
$ echo '
foo:
  bar: baz
' | dasel -r yaml "foo.prop"
Error: could not access map index: property not found: prop
```

how can i make dasel return an empty value and not to fail when accessing non existing property?

**Comment by TomWright** [comment]
That was actually a bug stopping errors being displayed for select commands, although errors should we written to stderr with stdout being blank.

**Reply by electriquo** [reply]
thank for the information, but is there a way to make dasel return an empty value (e.g. nil) and not to fail when accessing non-existing property?

**Reply by TomWright** [reply, marked as answer]
You can follow the property name with a `?` to allow dasel to continue without error: https://daseldocs.tomwright.me/functions/property#property-with-optional-field

**Reply by electriquo** [reply]
@TomWright Thanks you :)
